firedrake/mg/ufl_utils.py

```diff
@@ -104,6 +105,12 @@ def coarsen_form(form, self, coefficient_mapping=None):
     return form
 
 
+@coarsen.register(ufl.FormSum)
+def coarsen_formsum(form, self, coefficient_mapping=None):
+    return type(form)(*[(self(i, self, coefficient_mapping=coefficient_mapping), wi)
```

Comment:
Should we also coarsen the weights?

Reply:
At the moment weights are always constant, but here we generalize any future extensions